gpu/octree/src/cuda/approx_nsearch.cu

```diff
@@ -45,9 +47,51 @@
 
 
 namespace pcl { namespace device { namespace appnearest_search
-{   
+{
     using PointType = OctreeImpl::PointType;
-	
+
+	__host__ __device__  std::pair<uint3, std::uint8_t> nearestVoxel(const float3 query, const unsigned& level, const std::uint8_t& mask, const float3& minp, const float3& maxp, const uint3& index)
+	{
+		assert(mask != 0);
+		//identify closest voxel
+		float closest_distance = std::numeric_limits<float>::max();
+		unsigned closest_index = 0;
+		uint3 closest = make_uint3(0,0,0);
+
+		for (unsigned i = 0; i < 8; ++i)
+		{
+			if ((mask & (1<<i)) == 0)   //no child
+				continue;
+
+			const uint3 child = make_uint3(
+				(index.x << 1) + (i & 1),
+				(index.y << 1) + ((i>>1) & 1),
+				(index.z << 1) + ((i>>2) & 1));
+
+			//find center of child cell
+			const unsigned voxel_width = 1 << (level + 2);
```

Comment:
the name of this variable is misleading. this variable is not storing the width of the voxel. It's storing a scale factor which will directly affect the voxels' width.

Reply:
Yeah that's true, I think i originally meant width as in "width of the octree in voxels at current level", but that's really not what `voxel_width` naturally denotes. Would `no_of_voxels` or `voxels_per_side` be better? I can't seem to come up with a better name  :(

Reply:
`voxel_width_scale_factor`?